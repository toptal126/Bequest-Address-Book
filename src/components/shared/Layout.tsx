import { AppBar, Container, Stack, Toolbar, Typography } from "@mui/material";
import { FunctionComponent } from "react";
import { Link } from "react-router-dom";
interface LayoutProps {
  children: React.ReactNode;
}

const Layout: FunctionComponent<LayoutProps> = ({ children }) => {
  return (
    <Stack>
      <AppBar sx={{ backgroundColor: "#fff" }} elevation={0}>
        <Container>
          <Toolbar disableGutters>
            <img src={"/addressBook.png"} alt="Logo" width={40} />
            <Link data-testid="header-logo-link" to="/">
              <Typography
                data-testid="header-logo-text"
                variant="h5"
                noWrap
                sx={{
                  mx: 2,
                  display: { xs: "none", md: "flex" },
                  fontFamily: "monospace",
                  fontWeight: 700,
                  letterSpacing: ".3rem",
                  color: "#000",
                  textDecoration: "none",
                }}
              >
                Address Book
              </Typography>
            </Link>
          </Toolbar>
        </Container>
      </AppBar>
      <Toolbar
        sx={{
          marginBottom: "16px",
        }}
      />
      <Container
        data-testid="layout-container"
        sx={{
          display: "flex",
          flexDirection: "column",
        }}
      >
        {children}
      </Container>
    </Stack>
  );
};

export default Layout;
