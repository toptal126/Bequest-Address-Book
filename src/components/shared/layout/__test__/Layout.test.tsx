// import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Layout from "../../Layout";

describe("Layout", () => {
  test("Header logo link", () => {
    render(
      <BrowserRouter>
        <Layout>
          <h1>test</h1>
        </Layout>
      </BrowserRouter>
    );
    const element = screen.getByTestId("header-logo-link");
    expect(element).toHaveAttribute("href", "/");
  });

  test("Header logo text", () => {
    render(
      <BrowserRouter>
        <Layout>
          <h1>test</h1>
        </Layout>
      </BrowserRouter>
    );
    const element = screen.getByTestId("header-logo-text");
    expect(element).toHaveTextContent("Address Book");
  });

  test("Layout children", () => {
    render(
      <BrowserRouter>
        <Layout>
          <h1>test</h1>
        </Layout>
      </BrowserRouter>
    );
    const element = screen.getByTestId("layout-container");
    expect(element).toHaveTextContent("test");
  });

  test("Layout snapshot", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Layout>
          <h1>test</h1>
        </Layout>
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
