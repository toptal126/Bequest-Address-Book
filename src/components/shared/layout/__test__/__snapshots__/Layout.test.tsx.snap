// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Layout snapshot 1`] = `
<div
  className="css-nen11g-MuiStack-root"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1aodap3-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <img
          alt="Logo"
          src="/addressBook.png"
          width={40}
        />
        <a
          data-testid="header-logo-link"
          href="/"
          onClick={[Function]}
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-jgqr1i-MuiTypography-root"
            data-testid="header-logo-text"
          >
            Address Book
          </h5>
        </a>
      </div>
    </div>
  </header>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-12jwn1w-MuiToolbar-root"
  />
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1xon2ic-MuiContainer-root"
    data-testid="layout-container"
  >
    <h1>
      test
    </h1>
  </div>
</div>
`;
